lib/msf/core/session/provider/single_command_shell.rb

```diff
@@ -86,59 +86,55 @@ def shell_read_until_token(token, wanted_idx=0, timeout=10)
 
   def shell_command_token(cmd, timeout=10)
     if platform == 'windows'
-      output = shell_command_token_win32(cmd, timeout)
+      output = shell_command_token_base(cmd, timeout, '&')
     else
-      output = shell_command_token_unix(cmd, timeout)
+      output = shell_command_token_base(cmd, timeout, ';')
```

Comment:
Can we keep the `shell_command_token_win32` and `shell_command_token_unix`, updating them to call `shell_command_token_base` with the necessary value you have specified here?

For example
```ruby
def shell_command_token_win32(cmd, timeout)
  shell_command_token_base(cmd, timeout, '&')
end
```

This will help us avoid breaking Pro which does make use of these functions directly. It would also mean that the top two files could remain unchanged.

One more request, would you mind moving the two new functions under a `module_function` directive so they're protected.